Sources/Datadog/URLSessionAutoInstrumentation/Interception/TaskInterception.swift

```diff
@@ -26,6 +26,10 @@ internal class TaskInterception {
         self.identifier = UUID()
         self.request = request
         self.isFirstPartyRequest = isFirstParty
+        // TODO: RUMM-1064
+        // create AppStateListening.build()
```

Comment:
This would create `NotificationCenter` listener for every single request sent from the app. Isn't it better to share single listener among all requests? The global listener may also be needed for other components in Background Events Monitoring project.

Reply:
the worst case scenario with non-singleton can be ~100 simultaneous listeners and i added such test case, my measurement gives reasonable results (`100 -> 0.046sec`) even in this case.

i didn't prefer a singleton here as it will stay around even while not needed but singleton would work too.
if background monitoring requires a singleton, we can add `static var sharedHistory` and use it in `TaskInterception` and that will be it, shouldn't require major refactoring.

Reply:
i ensured main thread for getting current app state due to Main Thread Checker warning but now every `AppStateListener(...)` call leads to `main.sync { ... }`.
a singleton can solve this issue. i'm making the change.